### ISSUE TITLE
LLVM ERROR: Cannot select: 0x155f5770: v8i64 = X86ISD::PCMPEQ 0xc1cb780, 0x11db3070, simd_avx512bw.cpp:58:7

### ISSUE THREAD

**Issue by locus-x64** [llvm/llvm-project#111380]
While compiling one of my favorite targets, the linker command failed due to a signal. Please find the stack trace here.
```bash
Stack dump:
0.	Program arguments: /usr/lib/llvm-14/bin/ld.lld -pie -z relro --hash-style=gnu --build-id --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o ugrep /lib/x86_64-linux-gnu/Scrt1.o /lib/x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/12/crtbeginS.o -L/usr/lib -L/usr/lib -L/usr/lib -L/usr/lib -L/usr/lib -L/usr/lib -L/usr/lib -L/usr/lib -L/usr/lib -L/usr/lib -L../lzma/C -L/usr/lib/gcc/x86_64-linux-gnu/12 -L/usr/lib/gcc/x86_64-linux-gnu/12/../../../../lib64 -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib64 -L/usr/lib/llvm-14/bin/../lib -L/lib -L/usr/lib -plugin-opt=mcpu=x86-64 -plugin-opt=O2 -plugin-opt=new-pass-manager --whole-archive /usr/lib/llvm-14/lib/clang/14.0.0/lib/linux/libclang_rt.asan_static-x86_64.a --no-whole-archive --whole-archive /usr/lib/llvm-14/lib/clang/14.0.0/lib/linux/libclang_rt.asan-x86_64.a --no-whole-archive --dynamic-list=/usr/lib/llvm-14/lib/clang/14.0.0/lib/linux/libclang_rt.asan-x86_64.a.syms --whole-archive /usr/lib/llvm-14/lib/clang/14.0.0/lib/linux/libclang_rt.asan_cxx-x86_64.a --no-whole-archive --dynamic-list=/usr/lib/llvm-14/lib/clang/14.0.0/lib/linux/libclang_rt.asan_cxx-x86_64.a.syms --lto-legacy-pass-manager -mllvm=-load=/usr/local/bin/../lib/afl/SanitizerCoverageLTO.so --allow-multiple-definition ugrep-ugrep.o ugrep-cnf.o ugrep-glob.o ugrep-output.o ugrep-query.o ugrep-screen.o ugrep-stats.o ugrep-vkey.o ugrep-zopen.o -lpthread ../lib/libreflex.a -lviiz -lbrotlidec -lbrotlienc -lzstd -llzma -lz -lpcre2-8 /usr/local/bin/../lib/afl/afl-compiler-rt.o /usr/local/bin/../lib/afl/afl-llvm-rt-lto.o --dynamic-list=/usr/local/bin/../lib/afl/dynamic_list.txt -lstdc++ -lm --no-as-needed -lpthread -lrt -lm -ldl -lgcc_s -lgcc -lc -lgcc_s -lgcc /usr/lib/gcc/x86_64-linux-gnu/12/crtendS.o /lib/x86_64-linux-gnu/crtn.o
1.	Running pass 'Function Pass Manager' on module 'ld-temp.o'.
2.	Running pass 'X86 DAG->DAG Instruction Selection' on function '@_ZN6reflex21simd_nlcount_avx512bwERPKcS1_'
 #0 0x000074cc0323fd01 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xe3fd01)
 #1 0x000074cc0323da3e llvm::sys::RunSignalHandlers() (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xe3da3e)
 #2 0x000074cc03240236 (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xe40236)
 #3 0x000074cc01c42520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
 #4 0x000074cc01c969fc __pthread_kill_implementation ./nptl/./nptl/pthread_kill.c:44:76
 #5 0x000074cc01c969fc __pthread_kill_internal ./nptl/./nptl/pthread_kill.c:78:10
 #6 0x000074cc01c969fc pthread_kill ./nptl/./nptl/pthread_kill.c:89:10
 #7 0x000074cc01c42476 gsignal ./signal/../sysdeps/posix/raise.c:27:6
 #8 0x000074cc01c287f3 abort ./stdlib/./stdlib/abort.c:81:7
 #9 0x000074cc0317a723 (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xd7a723)
#10 0x000074cc03a60e31 (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x1660e31)
#11 0x000074cc03a602d8 llvm::SelectionDAGISel::SelectCodeCommon(llvm::SDNode*, unsigned char const*, unsigned int) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x16602d8)
#12 0x000074cc05a00c1a (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x3600c1a)
#13 0x000074cc03a58ecf llvm::SelectionDAGISel::DoInstructionSelection() (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x1658ecf)
#14 0x000074cc03a58594 llvm::SelectionDAGISel::CodeGenAndEmitDAG() (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x1658594)
#15 0x000074cc03a579c8 llvm::SelectionDAGISel::SelectAllBasicBlocks(llvm::Function const&) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x16579c8)
#16 0x000074cc03a55557 llvm::SelectionDAGISel::runOnMachineFunction(llvm::MachineFunction&) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x1655557)
#17 0x000074cc059f7a0d (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x35f7a0d)
#18 0x000074cc035c628e llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x11c628e)
#19 0x000074cc0337a390 llvm::FPPassManager::runOnFunction(llvm::Function&) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xf7a390)
#20 0x000074cc03381983 llvm::FPPassManager::runOnModule(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xf81983)
#21 0x000074cc0337af36 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xf7af36)
#22 0x000074cc04765acf (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x2365acf)
#23 0x000074cc04764cac llvm::lto::backend(llvm::lto::Config const&, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream> > > (unsigned int)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex&) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x2364cac)
#24 0x000074cc04759859 llvm::lto::LTO::runRegularLTO(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream> > > (unsigned int)>) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x2359859)
#25 0x000074cc04759053 llvm::lto::LTO::run(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream> > > (unsigned int)>, std::function<llvm::Expected<std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream> > > (unsigned int)> > (unsigned int, llvm::StringRef)>) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x2359053)
#26 0x00000000005ca433 lld::elf::BitcodeCompiler::compile() (/usr/lib/llvm-14/bin/ld.lld+0x5ca433)
#27 0x0000000000550206 void lld::elf::LinkerDriver::compileBitcodeFiles<llvm::object::ELFType<(llvm::support::endianness)1, true> >(bool) (/usr/lib/llvm-14/bin/ld.lld+0x550206)
#28 0x000000000054d05d lld::elf::LinkerDriver::link(llvm::opt::InputArgList&) (/usr/lib/llvm-14/bin/ld.lld+0x54d05d)
#29 0x0000000000541ac4 lld::elf::LinkerDriver::linkerMain(llvm::ArrayRef<char const*>) (/usr/lib/llvm-14/bin/ld.lld+0x541ac4)
#30 0x000000000053fd97 lld::elf::link(llvm::ArrayRef<char const*>, llvm::raw_ostream&, llvm::raw_ostream&, bool, bool) (/usr/lib/llvm-14/bin/ld.lld+0x53fd97)
#31 0x000000000048b235 (/usr/lib/llvm-14/bin/ld.lld+0x48b235)
#32 0x000000000048a998 main (/usr/lib/llvm-14/bin/ld.lld+0x48a998)
#33 0x000074cc01c29d90 __libc_start_call_main ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
#34 0x000074cc01c29e40 call_init ./csu/../csu/libc-start.c:128:20
#35 0x000074cc01c29e40 __libc_start_main ./csu/../csu/libc-start.c:379:5
#36 0x000000000048a4d5 _start (/usr/lib/llvm-14/bin/ld.lld+0x48a4d5)
```

**Comment by EugeneZelenko** [comment]
@locus-x64: Could you please try 19 or `main` branch? Could you please provide reproducer?

**Comment by RKSimon** [comment]
This might be a duplicate of #53842 (and #59909) - which was fixed by 2808743cbdcee806eecd7b8799cf776727dc574b

**Comment by locus-x64** [comment]
I just tried the latest version, and it worked for me. Thanks @EugeneZelenko and @RKSimon
